From 97689a3a7ac9aee9947cbeb852d7a8a743229ef5 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:27:05 -0400
Subject: [net] genetlink: remove superfluous assignment

Message-id: <833cdbb1d6bfc6beeae64e1a25db8a5c0ba971b3.1410528958.git.jbenc@redhat.com>
Patchwork-id: 93961
O-Subject: [RHEL7.1 net PATCH 23/23] genetlink: remove superfluous assignment
Bugzilla: 1140656
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1140656
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit 2f91abd4516da0cadfb955c963b16f96e2e6c07a
Author: Denis ChengRq <crquan@gmail.com>
Date:   Mon Jun 2 01:18:01 2014 -0700

    genetlink: remove superfluous assignment

    the local variable ops and n_ops were just read out from family,
    and not changed, hence no need to assign back.

    Validation functions should operate on const parameters and not
    change anything.

    Signed-off-by: Cheng Renquan <crquan@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netlink/genetlink.c b/net/netlink/genetlink.c
index a3ba3ca..76393f2 100644
--- a/net/netlink/genetlink.c
+++ b/net/netlink/genetlink.c
@@ -317,7 +317,7 @@ static void genl_unregister_mc_groups(struct genl_family *family)
  }
 }
 
-static int genl_validate_ops(struct genl_family *family)
+static int genl_validate_ops(const struct genl_family *family)
 {
  const struct genl_ops *ops = family->ops;
  unsigned int n_ops = family->n_ops;
@@ -337,10 +337,6 @@ static int genl_validate_ops(struct genl_family *family)
     return -EINVAL;
  }
 
- /* family is not registered yet, so no locking needed */
- family->ops = ops;
- family->n_ops = n_ops;
-
  return 0;
 }
 
-- 
1.7.1